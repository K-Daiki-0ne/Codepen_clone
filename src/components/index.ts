export { Editor } from './Editor/Editor';
