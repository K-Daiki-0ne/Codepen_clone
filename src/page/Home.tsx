import React, { useState, useEffect } from 'react';
import { useLocalStorage } from '../hook/useLocalStorage';

export const Home: React.FC = (): JSX.Element => {
  const [html, setHtml] = useLocalStorage('html', '');
  const [css, setCss] = useLocalStorage('css', '');
  const [js, setJs] = useLocalStorage('js', '');
  const [srcDoc, setSrcDoc] = useState<string>('');

  useEffect(() => {
    const timeout = setTimeout(() => {
      setSrcDoc(`
        <html>
          <body>${html}</body>
          <style>${css}</style>
          <script>${js}</script>
        </html>
      `)
    }, 250)

    return () => clearTimeout(timeout)
  }, [html, css, js])

  return (
    <div className='pane top-pane'>

    </div>
  )
}