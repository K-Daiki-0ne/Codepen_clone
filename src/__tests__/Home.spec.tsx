import React from 'react';
import { shallow } from 'enzyme';
import { Home } from '../page/Home';
import { Editor } from '../components/index'
import { unmountComponentAtNode } from "react-dom";

let container: any = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Home Component Test', () => {
  const wrapper = shallow(<Home />);

  test('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('rendering test', () => {
    const wrap = wrapper.find('div');
    
    test('Home Component', () => expect(wrap).toBeTruthy());

    describe('children component test', () => {
      const editor = wrap.find(Editor);
      expect(editor).toBeTruthy();
    });

    describe('iframe test', () => {
      const frame = wrap.find('iframe');
      expect(frame).toBeTruthy();
    });

  })
})