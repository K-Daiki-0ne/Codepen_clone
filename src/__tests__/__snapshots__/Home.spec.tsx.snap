// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `ShallowWrapper {}`;

exports[`Home Component Test snapshot test 1`] = `ShallowWrapper {}`;
